internal/core/cppapi_data.cpp

```diff
@@ -189,14 +193,8 @@ void fence(
         auto coordinate = transform_coordinate(x, y);
 
         auto validate_boundary = [&] (const int voxel, Axis const& axis) {
-            if(!axis.inrange(coordinate[voxel])){
-                const std::string coordinate_str =
-                    "(" +std::to_string(x) + "," + std::to_string(y) + ")";
-                throw std::runtime_error(
-                    "Coordinate " + coordinate_str + " is out of boundaries "+
-                    "in dimension "+ std::to_string(voxel)+ "."
-                );
-            }
+            if(!axis.inrange(coordinate[voxel]))
```

Comment:
I think using { } is always safer for "ifs". 
From experience it is too easy to not notice when adding second line, for example for debugging, and then wonder why my code doesn't work the way I want it to :smile:

Reply:
Sure, 😂